### PR TITLE
CB2-5365 apply date formatting

### PR DESCRIPTION
**Pull request by Ian-Hodges**
## Description
The responses for vehicle and testResults contain datetimes with trailing zeros. The date format should be yyyy-MM-dd HH:mm:ss. 

The trailing zeros are coming from mysql's JSON_OBJECT. For some reason it formats dates with .000000 for the milliseconds.

Jira ticket: [CB2-5365](https://dvsa.atlassian.net/browse/CB2-5365).

#### Develop Response
image
#### VTMDEV-1 Response
image
#### Diff
The only difference between develop and VTMDEV-1 is the removal of the trailing zeros.
image

#### Diff
The only difference between develop and VTMDEV-1 is the removal of the trailing zeros.
image

### CODE REVIEW

**Review by naathanbrown — APPROVED** [review]

**Review by aaron-downing — DISMISSED** [review]

File: src/app/databaseService.ts [hunk 1]
```diff
@@ -139,9 +139,9 @@ async function getTestResultDetails(
   const testResultQueryResult = queryResult[0][0] as TestResultQueryResult;
 
   if (
-    testResultQueryResult === undefined ||
-    testResultQueryResult.id === undefined ||
-    testResultQueryResult.result === undefined
+    testResultQueryResult === undefined
```
Comment (Sul-Aga-kainos) [inline src/app/databaseService.ts:142]: Maybe add a function to include these checks like  `isResultDefined` or something similar

File: src/app/queries/technicalRecord/technicalRecordQuery.ts [hunk 2]
```diff
@@ -1,105 +1,107 @@
-const TECHNICAL_RECORD_QUERY = 'SELECT `t`.`id`, JSON_OBJECT('
-  + "'recordCompleteness', `t`.`recordCompleteness`, "
-  + "'createdAt', `t`.`createdAt`, "
-  + "'lastUpdatedAt', `t`.`lastUpdatedAt`, "
+import { dateFormat } from '../../../constants';
+
+const TECHNICAL_RECORD_QUERY = 'SELECT t.id, JSON_OBJECT('
```
[the remark this answers is not in the document]
Comment (Ian-Hodges) [reply src/app/queries/technicalRecord/technicalRecordQuery.ts:3]: I have used string templates where I added the dateFormat variable. I'm not sure where else I'd use it?
Are you asking why we don't include the variable directly in the query?

File: src/app/databaseService.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ian-Hodges) [reply src/app/databaseService.ts:142]: This file was only changed by the linter.
A function would work. It's only in a couple of places so, if I were to change it, I probably just go with:
`if (
    vehicleDetailsResult?.id === undefined
    || vehicleDetailsResult?.result === undefined
  )`

[the remark this answers is not in the document]
Comment (Sul-Aga-kainos) [reply src/app/databaseService.ts:142]: yeah that would work

File: src/app/queries/technicalRecord/technicalRecordQuery.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Sul-Aga-kainos) [reply src/app/queries/technicalRecord/technicalRecordQuery.ts:3]: yeah what I meant is wrapping the whole query in ticks (string template) and get rid of + on every new line. I think the query will be more readable

[the remark this answers is not in the document]
Comment (Ian-Hodges) [reply src/app/queries/technicalRecord/technicalRecordQuery.ts]: Done. It does look a lot better.

File: src/app/queries/technicalRecord/vehicleDetailsQuery.ts [hunk 3]
```diff
@@ -1,14 +1,16 @@
-const baseQuery = 'SELECT `v`.`id`, JSON_OBJECT('
-  + "'system_number', `v`.`system_number`, "
-  + "'vin', `v`.`vin`, "
-  + "'vrm_trm', `v`.`vrm_trm`, "
-  + "'trailer_id',`v`.`trailer_id`, "
-  + "'createdAt', `v`.`createdAt`"
+import { dateFormat } from '../../../constants';
+
+const baseQuery = 'SELECT v.id, JSON_OBJECT('
```
[the remark this answers is not in the document]
Comment (Ian-Hodges) [reply src/app/queries/technicalRecord/vehicleDetailsQuery.ts]: Done. It does look a lot better.

File: src/app/databaseService.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ian-Hodges) [reply src/app/databaseService.ts]: Simplification and unit tests added.

Comment (Ian-Hodges) [reply src/app/databaseService.ts]: Simplification and unit tests added.

**Review by aaron-downing — APPROVED** [review]

**Review by kpearmanbjss — APPROVED** [review]

**Pull request merged**